v2/internal/pkg/operator/filtered_collector.go

```diff
@@ -273,7 +273,17 @@ func (o FilterCollector) collectOperator( //nolint:cyclop // TODO: this needs fu
 		rebuiltTag = tag
 	}
 
+	// OCPBUGS-52470
+	// check if the original operator was mirrored by digest
 	componentName := imgSpec.ComponentName() + "." + result.Digest
+	if imgSpec.IsImageByDigestOnly() && o.Opts.IsMirrorToDisk() {
+		tag, err := digestOfFilter(op)
+		if err != nil {
+			return v2alpha1.CatalogFilterResult{}, err
```

Comment:
Can we add more context to this error? `digestOfFilter` can fail during `json.Marshal` and marshaling errors don't contain enough context to point out where things were wrong.
```suggestion
			return v2alpha1.CatalogFilterResult{}, fmt.Errorf("failed to get filter digest for operator mirrored by digest: %w", err)
```